engine/src/flutter/impeller/entity/contents/content_context.h

```diff
@@ -521,7 +522,13 @@ class ContentContext {
     void SetDefault(const ContentContextOptions& options,
                     std::unique_ptr<PipelineHandleT> pipeline) {
       default_options_ = options;
-      Set(options, std::move(pipeline));
+      if (pipeline) {
+        Set(options, std::move(pipeline));
+      }
+    }
+
+    void SetDefaultDescriptor(std::optional<PipelineDescriptor> desc) {
+      desc_ = std::move(desc);
```

Comment:
Should this just be another argument to SetDefault?

Reply:
Its a bit more fiddly than that unfortunately